src/main/java/com/kobylynskyi/graphql/codegen/model/MappingContext.java

```diff
@@ -275,7 +280,44 @@ public GeneratedInformation getGeneratedInformation() {
     }
 
     public Set<String> getEnumImportItSelfInScala() {
+        // Only for scala
+        if (GeneratedLanguage.SCALA.equals(this.config.getGeneratedLanguage()) && enumImportItSelfInScala == null) {
+            enumImportItSelfInScala = this.document.getEnumDefinitions().parallelStream().map(this::getModelClassNameWithPrefixAndSuffix).collect(Collectors.toSet());
+        }
         return enumImportItSelfInScala;
     }
 
+    public Map<String, Set<String>> getParentInterfaceProperties() {
```

Comment:
I might refactor this in the future to have a separate `MappingContext`s for diff languages.
But for now looks ok!

Reply:
Today I made a improvement for kotlin / Scala (but not implement in Scala)